pkg/resources/external_function.go

```diff
@@ -473,63 +445,115 @@ func ReadExternalFunction(d *schema.ResourceData, meta interface{}) error {
 				}
 
 				if err := d.Set("header", headers); err != nil {
-					return err
+					return diag.Errorf("error setting return_behavior: %v", err)
 				}
 			}
 		case "context_headers":
-			if desc.Value.Valid && desc.Value.String != "null" {
+			if row.Value != "" && row.Value != "null" {
 				// Format in Snowflake DB is: ["CONTEXT_FUNCTION_1","CONTEXT_FUNCTION_2"]
-				contextHeaders := strings.Split(strings.ReplaceAll(strings.ReplaceAll(strings.ReplaceAll(desc.Value.String, "[", ""), "]", ""), "\"", ""), ",")
-
+				contextHeaders := strings.Split(strings.Trim(row.Value, "[]"), ",")
+				for i, v := range contextHeaders {
+					contextHeaders[i] = strings.Trim(v, "\"")
+				}
 				if err := d.Set("context_headers", contextHeaders); err != nil {
-					return err
+					return diag.Errorf("error setting context_headers: %v", err)
 				}
 			}
 		case "max_batch_rows":
-			if desc.Value.String != "not set" {
-				i, err := strconv.ParseInt(desc.Value.String, 10, 64)
+			if row.Value != "not set" {
+				maxBatchRows, err := strconv.ParseInt(row.Value, 10, 64)
 				if err != nil {
-					return err
+					return diag.Errorf("error parsing max_batch_rows: %v", err)
 				}
-
-				if err := d.Set("max_batch_rows", i); err != nil {
-					return err
+				if err := d.Set("max_batch_rows", maxBatchRows); err != nil {
+					return diag.Errorf("error setting max_batch_rows: %v", err)
 				}
 			}
 		case "compression":
-			if err := d.Set("compression", desc.Value.String); err != nil {
-				return err
+			if err := d.Set("compression", row.Value); err != nil {
+				return diag.Errorf("error setting compression: %v", err)
 			}
 		case "body":
-			if err := d.Set("url_of_proxy_and_resource", desc.Value.String); err != nil {
-				return err
+			if err := d.Set("url_of_proxy_and_resource", row.Value); err != nil {
+				return diag.Errorf("error setting url_of_proxy_and_resource: %v", err)
 			}
 		case "language":
 			// To ignore
 		default:
-			log.Printf("[WARN] unexpected external function property %v returned from Snowflake", desc.Property.String)
+			log.Printf("[WARN] unexpected external function property %v returned from Snowflake", row.Property)
 		}
 	}
 
 	return nil
 }
 
-// DeleteExternalFunction implements schema.DeleteFunc.
-func DeleteExternalFunction(d *schema.ResourceData, meta interface{}) error {
+func UpdateContextExternalFunction(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
 	db := meta.(*sql.DB)
-	externalFunctionID, err := externalFunctionIDFromString(d.Id())
-	if err != nil {
-		return err
+	client := sdk.NewClientFromDB(db)
+	parts := strings.Split(d.Id(), helpers.IDDelimiter)
+	if len(parts) != 4 {
+		return diag.Errorf("external function ID is not valid: %s", d.Id())
 	}
+	databaseName := parts[0]
+	schemaName := parts[1]
+	name := parts[2]
+	argtypes := parts[3]
+
+	arguments := make([]sdk.DataType, 0)
+	if argtypes != "" {
+		argTypes := strings.Split(argtypes, "-")
```

Comment:
this split logic is repeated, please extract it

Reply:
fixed